### PR TITLE
Expanded the section on closures in TRPL.

### PR DESCRIPTION
**Pull request by js-ojus**
The section now illustrates a function that accepts two closures, showing how each gets its own distinct type, even when both have the same apparent signature.

### CODE REVIEW

File: src/doc/trpl/closures.md [hunk 1]
```diff
@@ -181,6 +174,36 @@ fn main() {
 
 Doing this is not particularly common, but it's useful every once in a while.
 
+Before we move on, let us look at a function that accepts two closures.
+
+```{rust}
+fn compose<F, G>(x: i32, f: F, g: G) -> i32
+    where F: Fn(i32) -> i32, G: Fn(i32) -> i32 {
```
Comment (sinistersnare) [inline src/doc/trpl/closures.md]: What is the style guideline for brackets on where-clause? newline or same line?

Comment (steveklabnik) [inline src/doc/trpl/closures.md]: @aturon ?

File: src/doc/trpl/closures.md [hunk 2]
```diff
@@ -181,6 +174,39 @@ fn main() {
...
+
+You might ask yourself: why do we need to introduce two type
+parameters `F` and `G` here?  Evidently, both `f` and `g` have the
+same signature: `Fn(i32) -> i32`.
+
+That is because in Rust each closure has its own unique type.
+So, not only do closures with different signatures have different types,
+but different closures with the *same* signature have *different*
+types, as well!
+You can think of it this way: the behaviour of a closure is part of its type.
```
Comment (steveklabnik) [inline src/doc/trpl/closures.md]: If this was intended to be a separate paragraph (and it should be), you need another newline here.

Comment (steveklabnik) [inline src/doc/trpl/closures.md]: (Also, no `u` in 'behavior', as much as I prefer to the British spelling. :)

**Comment by steveklabnik** [comment]
This has a toooooon of merge commits, so it's gonna need rebased and squashed.

**Comment by js-ojus** [comment]
@steveklabnik Sure.  Could you close this request please?  I shall submit another.  Thanks.

File: src/doc/trpl/closures.md (hunk 1 above)
Comment (aturon) [inline src/doc/trpl/closures.md]: When wrapping a `where` like this I usually see the `{` on the next line, but there's no official guideline.

**Pull request closed without merging**